CMakeLists.txt

```diff
@@ -3,15 +3,20 @@ project(poco_vendor VERSION "1.1.1")
 
 # Can work with poco 1.4.1p1 (earliest to use recursive mutexes on Linux)
 # 1.6.1 is the first version to ship with PocoConfigVersion.cmake
-find_package(Poco "1.6.1" COMPONENTS Foundation QUIET)
+if(NOT FORCE_BUILD_POCO)
```

Comment:
This doesn't require a new CMake option (which should by the way be declared with `option(...)`). The same can be achieved with passing `CMAKE_DISABLE_FIND_PACKAGE_Poco`.

Reply:
I think we do have to be more precise than that. Taking a look at how `poco_vendor` is used, e.g.

https://github.com/ros2/rosidl_typesupport/blob/master/rosidl_typesupport_c/CMakeLists.txt#L19

The `find_package(poco_vendor)` call provides `FindPoco.cmake`, rather than Poco itself. If you globally set `-DCMAKE_DISABLE_FIND_PACKAGE_Poco=ON`, then upstream packages can't find our built version of Poco either.

That being considered, I will update this to use an `option`

Reply:
This pattern seems generic to the vendored packages, so TinyXML runs into the same problem.

This is all unless I am missing a way to pass a CMake arg only to a specific package in the colcon build?

Reply:
Checking on the [CMAKE_DISABLE_FIND_PACKAGE_\<package\>](https://cmake.org/cmake/help/v3.5/command/find_package.html) docs, I agree with @emersonknapp that they won't work in this case without changing every consumer package to either override the call being disabled or use a non-find_package() mechanism for the dependency, a change which would have to be reverted when the vendor packages are retired. There might be something nefarious we could do to unset that argument when our vendor package kicks in, but I'm not sure that would work and if not done carefully I think it might result in the vendored package being rebuilt for every package that runs `find_package(foo_vendor)`.

Given that we're aiming to retire all of the vendor packages by the 20.04 release when :crossed_fingers: all our dependencies will be available with appropriate version numbers via some system packaging mechanism {apt, brew, TBD (Windows)}. I find the added options for forcing vendor packages to build an appropriate 1-year solution and think that unless a cheap or free standard solution is presented, as the CMAKE_DISABLE_FIND_PACKAGE_ would have been if it worked for our case, then the engineering time we'd spend iterating on this is better spent preparing for the future when we do not have vendor packages as a mechanism to perform from-source builds for some system dependencies.

Reply:
If I had to bikeshed a little, I could see arguments both for using one flag for all ROS 2 vendor packages that support building from source or using package-specific ones as is currently proposed, but don't have strong feelings either way.

Reply:
@dirk-thomas and I had a discussion about this and using a single option for all vendor packages, `FORCE_BUILD_VENDOR_PKG` was proposed, is the preferred route. While it still requires the use of a new option, it keeps our mixin from needing to reference every specific vendor package.

@emersonknapp I don't see any reason why a consistent option wouldn't work for the libcxx builds but if there's something I'm missing please let me know, otherwise I think proceeding with the switch will get all these vendor package PRs ready for their next review.

Reply:
Ok - I will move it back to the global option - should I be defining it as an option in all of the vendor packages?

Reply:
>  I will move it back to the global option

I've been context switching on these so much I don't recall that this previously used a shared variable. Thanks for iterating as we yo-yo through different goals and constraints.

As far as defining the option everywhere. colcon builds each package in a separate CMake context so I believe it should be defined as an option for each of them. @wjwwood can I trade you a cola for a CMake consult?

Reply:
> As far as defining the option everywhere. colcon builds each package in a separate CMake context so I believe it should be defined as an option for each of them.

That is correct.

Was there a different question or just that?

Reply:
Nope, just verifying that using the `option()` directive in each package was the right way to go :+1: